app/scripts/views/WorkspaceView.js

```diff
@@ -551,6 +552,84 @@ define(['backbone', 'Workspace', 'ConnectionView', 'MarqueeView', 'NodeViewTypes
 
     },
 
+    watchNodeViewEvents: function(nodeView) {
+        this.listenTo(nodeView, 'end-port-conn', this.endPortConnection);
+        this.listenTo(nodeView, 'update-connections', function() {
+            this.cleanup().updateConnections();
+        });
+
+        this.listenTo(nodeView, 'request-open-definition', function(id) {
+            this.model.app.openWorkspace( id );
+        });
+
+        if (nodeView.changeVisibility) {
+            nodeView.listenTo(this.model, 'change:current', nodeView.changeVisibility.bind(nodeView, this.model));
+            nodeView.listenTo(nodeView.model, 'change:visible', nodeView.changeVisibility.bind(nodeView, this.model));
+        }
```

Comment:
passing the workspace into node view's method which doesn't know about it anymore